client/http-client/src/tests.rs

```diff
@@ -42,7 +41,33 @@ async fn method_call_works() {
 		.await
 		.unwrap()
 		.unwrap();
-	assert_eq!(JsonValue::String("hello".into()), result);
+	assert_eq!("hello", &result);
+}
+
+#[tokio::test]
+async fn method_call_with_wrong_id() {
+	let exp = "id as string";
+	let server_addr =
+		http_server_with_hardcoded_response(ok_response(exp.into(), Id::Num(0))).with_default_timeout().await.unwrap();
```

Comment:
To clarify, this fails because the server is configured to return numeric IDs and the client expects string IDs?

(would "wrong_id_kind" perhaps be a more precise test name?)

Reply:
exactly, thanks good point